### DISCUSSION TITLE
Ghost data gone on container re-creation

### DISCUSSION THREAD

**Discussion by RaicuRobert** [docker-library/ghost#365] [General]
I've followed the instructions on https://hub.docker.com/_/ghost/ and created a container with a volume as such 
`docker run -d --name some-ghost -e NODE_ENV=development -v some-ghost-data:/var/lib/ghost/content ghost`

Later on, I deleted the container to do some changes understanding that the volume has my data saved under the "/data" folder, but that is empty.

When I created the container again, all tables were recreated and all the data was gone.
Did I get something wrong?

**Comment by RaicuRobert** [comment]
I've found "ghost-dev.db" at the' /var/lib/ghost/current/content/data' path.
I've lost a few days' worths of articles 😥

**Comment by tianon** [comment]
Did the volume get deleted while the container was stopped? (`docker system prune --volumes`, etc)

**Reply by RaicuRobert** [reply]
The volume did not get deleted but, inside it, the '/data' folder is empty.
Mounting it back does nothing, as expected.
There are other files there but no db file.

After I created a new container, I navigated its folders and found the DB file at the wrong path, in '/var/lib/ghost/current/content/data'.
It was not in the mounted folder specified in '/var/lib/ghost/content'

**Comment by yosifkit** [comment]
Ah, the removal of `sqlite` for non-dev strikes again. It looks like https://github.com/docker-library/ghost/pull/323#issuecomment-1212554428 is partially needed even in `development` mode. So this works to get the file into the volume:
```console
$ docker run -d --name some-ghost \
-e NODE_ENV=development \
-e "database__connection__filename=/var/lib/ghost/content/data/ghost.db" \
-v some-ghost-data:/var/lib/ghost/content \
ghost:5
```

**Reply by RaicuRobert** [reply]
Thanks!
I've read a bit about the changes made in v5 and said it would be fun to start a blog.
I did not expect to run into a config issue like this straight out the gate :)).

What would be the best course of action here?
Since it's not recommend to use SQLite anymore, would it be better to recommend the docker compose option only?
Maybe leave the SQLite option at the end with a warning and the CMD you wrote.

**Comment by tianon** [comment]
Yeah, I think we should probably update our examples to stop even
recommending development mode and push everyone to use a "proper" SQL
database. 🙈
